drivers/video/ov5640.c

```diff
@@ -573,12 +659,236 @@ static int ov5640_stream_stop(const struct device *dev)
 	return ov5640_write_reg(&cfg->i2c, SYS_CTRL0_REG, SYS_CTRL0_SW_PWDN);
 }
 
+#define TEST_PATTERN_ENABLE  BIT(7)
+#define TEST_PATTERN_ROLLING BIT(6)
+#define TEST_PATTERN_BAR     (0 << 0)
+#define TEST_PATTERN_SQUARE  (2 << 0)
+
+static const uint8_t test_pattern_val[] = {
+	0,
+	TEST_PATTERN_ENABLE | TEST_PATTERN_BAR | (1 << 2),
+	TEST_PATTERN_ENABLE | TEST_PATTERN_BAR | (1 << 2) | TEST_PATTERN_ROLLING,
+	TEST_PATTERN_ENABLE | TEST_PATTERN_SQUARE,
+	TEST_PATTERN_ENABLE | TEST_PATTERN_SQUARE | TEST_PATTERN_ROLLING,
+};
+
+static int ov5640_set_ctrl_test_pattern(const struct device *dev, uint8_t value)
+{
+	const struct ov5640_config *cfg = dev->config;
+
```

Comment:
Should the boundary of `value` be checked or is it the responsibility of the caller?

```suggestion
	if (value >= ARRAY_SIZE(test_pattern_val)) {
		return -EINVAL;
	}
```

Reply:
The USB3CV validation suite for USB Video Class will actively push controls out of their range to see if the device is rejecting them.

If it is the caller's responsibility to check the boundaries, some other mechanism can be used though, i.e. querying the MIN/MAX (upcoming PR for this).

Reply:
It does not seem resolved to me? value should be checked.